test/test_jit.py

```diff
@@ -1192,6 +1196,46 @@ def observe(x, name):
         self.assertEqual(value_stats['p'][1], x2 + y2)
         self.assertEqual(value_stats['z'][1], x2 - y2)
 
+    def test_insert_param_observers(self):
+        x1 = torch.ones([1, 1, 1, 1])
+
+        class testModule(torch.jit.ScriptModule):
+            def __init__(self):
+                super(testModule, self).__init__()
+                self.conv1 = nn.Conv2d(1, 1, 1, 1)
+                self.conv1.weight.data.fill_(1.0)
+                self.conv1.bias.data.fill_(0.01)
+
+            @torch.jit.script_method
+            def forward(self, x):
+                x = self.conv1(x)
+                return x
+
+        trace = testModule()
+
+        # Custom observer function
+        value_stats = {}
+
+        class ObserverModule:
+            def observer(self, x, name):
+                if name not in value_stats:
+                    value_stats[name] = []
+                value_stats.update({name : x})
+                return x
+        observerObj = ObserverModule()
+
+        # Insert observers for parameters
+        observer_dict = {'__main__': {'param': observerObj}
```

Comment:
I don't think we need to insert observers for parameters. It's easier to just iterate over them on a python level and be done. Especially taking into account how specialized `InsertObserver` pass becomes with this change: basically it's unusable anywhere except in the quantization use-case.

Reply:
I agree its doable in python. This was a concious choice so value_stats dict is updated at one level. Else observer logic will be scattered between c++ and python layer for parameters and activations. Special case is not really specific to quantization but more between activations and parameters. Given we will be migrating to module level for our quantization passes, my opinion is to keep local. cc @dzhulgakov  @raghu

Reply:
the only argument for the current approach is that not all params might be used in the module and this logic quantizes only them. Not sure how much it matters in practice.

If you're worried about py/c++ split - nothing prevents you from iterating over parameters in C++ too :)